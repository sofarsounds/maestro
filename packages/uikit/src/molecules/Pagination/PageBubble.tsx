import React from 'react';

import Badge from '../../atoms/Badge';

interface BubbleProps {
  isActive: boolean;
  page: number;
  onClick: () => void;
}

const PageBubble: React.SFC<BubbleProps> = ({ isActive, page, onClick }) =>
  isActive ? (
    <Badge colour="primary">{page}</Badge>
  ) : (
    <Badge as="button" clickable onClick={onClick}>
      {page}
    </Badge>
  );

export default PageBubble;
