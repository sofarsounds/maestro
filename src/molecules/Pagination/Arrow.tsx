import React from 'react';

import Badge from '../../atoms/Badge';
import Icon from '../../atoms/Icon';

interface ArrowProps {
  onClick: () => void;
  disabled?: boolean;
  'data-qaid'?: string;
}

const renderArrow = (
  disabled: boolean = false,
  onClick: any,
  icon: string,
  qaId: string = ''
) => (
  <Badge
    as="button"
    disabled={disabled}
    clickable
    onClick={onClick}
    data-qaid={qaId}
  >
    <Icon color={disabled ? 'blueSmoke' : undefined} name={icon} />
  </Badge>
);

export const ArrowLeft: React.SFC<ArrowProps> = ({
  disabled,
  onClick,
  'data-qaid': qaId
}) => renderArrow(disabled, onClick, 'chevronLeft', qaId);

export const ArrowRight: React.SFC<ArrowProps> = ({
  disabled,
  onClick,
  'data-qaid': qaId
}) => renderArrow(disabled, onClick, 'chevronRight', qaId);
