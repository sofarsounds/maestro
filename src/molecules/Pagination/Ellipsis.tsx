import React from 'react';

import Icon from '../../atoms/Icon';
import Badge from '../../atoms/Badge';

const Ellipsis: React.SFC = () => (
  <Badge>
    <Icon name="ellipsisHorizontal" />
  </Badge>
);

export default Ellipsis;
