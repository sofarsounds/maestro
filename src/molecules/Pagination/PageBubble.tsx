import React from 'react';
import styled, { css } from '../../lib/styledComponents';

import Badge from '../../atoms/Badge';

interface BubbleProps {
  isActive: boolean;
  page: number;
  onClick: () => void;
  qaId?: string;
}

const BadgeWithSpacing = styled(Badge)`
  ${({ theme }) => css`
    margin: 0 ${theme.ruler[1]}px;
  `}
`;

const PageBubble: React.SFC<BubbleProps> = ({
  isActive,
  page,
  onClick,
  qaId = ''
}) =>
  isActive ? (
    <BadgeWithSpacing color="primary">{page}</BadgeWithSpacing>
  ) : (
    <BadgeWithSpacing
      as="button"
      clickable
      onClick={onClick}
      data-qaid={`${qaId}-page-${page}`}
    >
      {page}
    </BadgeWithSpacing>
  );

export default PageBubble;
