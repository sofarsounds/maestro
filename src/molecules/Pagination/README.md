# Pagination

To implement a `Pagination` into your project you'll need to add this import
```js
import { Pagination } from '@sofarsounds/maestro'
```

After adding the import you can use the Pagination simply like this
```html
<Pagination
  totalRecords={400}
  perPage={10}
  currentPage={4}
  onPageChange={p => console.log('new page', p)}
/>
```

## Props

| Name             | Type       | Default         | Description                      |
| :------------    | :-----     | :-------------- | :------------------------------- |
| **totalRecords** | `number`   |                 | Number of total records available
| **onPageChange** | `Function` |                 | Callback when page is being changed. Returns new page number
| perPage          | `number`   | `8`             | Customise how many items are being display per page
| currentPage      | `number`   | `1`             | Set the current page
| data-qaid        | `string`   |                 | Optional prop for testing purposes

