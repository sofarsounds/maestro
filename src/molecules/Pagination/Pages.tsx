import React from 'react';

import PageBubble from './PageBubble';

interface PagesProps {
  displayPages: number;
  currentPage: number;
  enlargement?: number;
  onClick: (p: number) => void;
  qaId?: string;
}

const Pages: React.SFC<PagesProps> = ({
  displayPages,
  currentPage,
  enlargement = 1,
  onClick,
  qaId = ''
}) => (
  <>
    {[...Array(displayPages)].map((_, page: number) => (
      <PageBubble
        key={page + enlargement}
        page={page + enlargement}
        isActive={currentPage === page + enlargement}
        onClick={() => onClick(page + enlargement)}
        qaId={qaId}
      />
    ))}
  </>
);

export default Pages;
