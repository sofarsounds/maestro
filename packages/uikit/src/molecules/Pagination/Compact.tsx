import React from 'react';

import PageBubble from './PageBubble';
import Pages from './Pages';
import Ellipsis from './Ellipsis';

interface CompactProps {
  currentPage: number;
  totalPages: number;
  onClickPage: (p: number) => void;
}

const Compact: React.SFC<CompactProps> = ({
  currentPage,
  totalPages,
  onClickPage
}) => {
  if (currentPage > totalPages - 4) {
    return (
      <>
        <PageBubble
          page={1}
          isActive={currentPage === 1}
          onClick={() => onClickPage(1)}
        />
        <Ellipsis />
        <Pages
          displayPages={5}
          currentPage={currentPage}
          enlargement={totalPages - 4}
          onClick={onClickPage}
        />
      </>
    );
  }

  if (currentPage < 5) {
    return (
      <>
        <Pages
          displayPages={5}
          currentPage={currentPage}
          onClick={onClickPage}
        />
        <Ellipsis />
        <PageBubble
          page={totalPages}
          isActive={currentPage === totalPages}
          onClick={() => onClickPage(totalPages)}
        />
      </>
    );
  }

  return (
    <>
      <PageBubble
        page={1}
        isActive={currentPage === 1}
        onClick={() => onClickPage(1)}
      />
      <Ellipsis />
      <Pages
        displayPages={3}
        currentPage={currentPage}
        enlargement={currentPage - 1}
        onClick={onClickPage}
      />
      <Ellipsis />
      <PageBubble
        page={totalPages}
        isActive={currentPage === totalPages}
        onClick={() => onClickPage(totalPages)}
      />
    </>
  );
};

export default Compact;
