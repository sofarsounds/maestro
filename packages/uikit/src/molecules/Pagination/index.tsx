import React from 'react';
import styled from '../../lib/styledComponents';

import CompactPages from './Compact';
import Pages from './Pages';
import { ArrowLeft, ArrowRight } from './Arrow';

const PaginationWrapper = styled.div`
  display: flex;
`;

export interface PaginationProps {
  currentPage?: number;
  onPageChange: (p: number) => void;
  totalRecords: number;
  perPage?: number;
  'data-qaid'?: string;
}

const MAX_PAGES = 5;
const PER_PAGE = 8;

export default class Pagination extends React.Component<PaginationProps> {
  public static DEFAULT_PER_PAGE = PER_PAGE;

  public handlePageChange = (page: number) => {
    this.setState({ currentPage: page });
    this.props.onPageChange(page);
  };

  public render() {
    const {
      currentPage = 1,
      totalRecords,
      'data-qaid': qaId = ''
    } = this.props;

    const perPage = this.props.perPage || PER_PAGE;
    const totalPages = Math.ceil(totalRecords / perPage);

    if (!totalRecords) {
      return null;
    }

    if (totalPages <= 1) {
      return null;
    }

    return (
      <PaginationWrapper data-qaid={qaId}>
        <ArrowLeft
          disabled={currentPage <= 1}
          onClick={() => this.handlePageChange(currentPage - 1)}
          data-qaid={`${qaId}-prev`}
        />

        {totalPages <= MAX_PAGES ? (
          <Pages
            displayPages={totalPages}
            currentPage={currentPage}
            onClick={this.handlePageChange}
          />
        ) : (
          <CompactPages
            currentPage={currentPage}
            totalPages={totalPages}
            onClickPage={this.handlePageChange}
          />
        )}

        <ArrowRight
          disabled={currentPage >= totalPages}
          onClick={() => this.handlePageChange(currentPage + 1)}
          data-qaid={`${qaId}-next`}
        />
      </PaginationWrapper>
    );
  }
}
