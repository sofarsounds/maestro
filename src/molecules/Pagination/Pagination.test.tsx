import React from 'react';

import { mountWithTheme } from '../../test';
import Pagination from './index';
import Compact from './Compact';
import PageBubble from './PageBubble';

const mockClick = jest.fn();

const setup = (currentPage: number, totalRecords: number, perPage?: number) =>
  mountWithTheme(
    <Pagination
      currentPage={currentPage}
      totalRecords={totalRecords}
      perPage={perPage}
      onPageChange={mockClick}
    />
  );

describe('<Pagination />', () => {
  beforeEach(() => jest.clearAllMocks());

  it('renders correctly', () => {
    expect(setup(3, 80)).toMatchSnapshot();
  });

  describe('Number of Pages calculation', () => {
    it('it calculates the number of pages with default settings', () => {
      const wrapper = setup(1, 80);

      expect(
        wrapper
          .find(PageBubble)
          .last()
          .props().page
      ).toBe(10);
    });

    it('it calculates the number of pages with custom perPage', () => {
      const wrapper = setup(1, 80, 5);

      expect(
        wrapper
          .find(PageBubble)
          .last()
          .props().page
      ).toBe(16);
    });
  });

  it("does not render <Compact /> if there's less than 5 pages", () => {
    const wrapper = setup(1, 5);
    expect(wrapper.find(Compact)).toHaveLength(0);
  });

  it("renders <Compact /> if there's more than 5 pages", () => {
    const wrapper = setup(1, 400);
    expect(wrapper.find(Compact)).toHaveLength(1);
  });

  it('triggers a callback when the user changes the page', () => {
    const wrapper = setup(1, 80);

    wrapper
      .find(PageBubble)
      .at(2)
      .simulate('click');
    expect(mockClick).toHaveBeenCalled();
  });
});
