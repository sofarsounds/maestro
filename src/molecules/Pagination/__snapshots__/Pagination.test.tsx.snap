// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> renders correctly 1`] = `
.c1 {
  width: 36px;
  height: 36px;
  margin: 0;
  padding: 0;
  font-size: 14px;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  border: none;
  border-radius: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c1:hover {
  background-color: rgba(17,17,17,0.1);
}

.c4 {
  width: 36px;
  height: 36px;
  margin: 0;
  padding: 0;
  font-size: 14px;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  border: none;
  border-radius: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #039842;
  color: #FFFFFF;
  font-weight: 600;
}

.c5 {
  width: 36px;
  height: 36px;
  margin: 0;
  padding: 0;
  font-size: 14px;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  border: none;
  border-radius: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3 {
  margin: 0 4px;
}

.c2 {
  font-family: 'sofarsounds-icon-font' !important;
  speak: none;
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;
  line-height: 1;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c2:before {
  color: #000000;
}

.c2.icon-facebook:before {
  content: '\\e920';
}

.c2.icon-instagram:before {
  content: '\\e929';
}

.c2.icon-spotify:before {
  content: '\\e937';
}

.c2.icon-twitter:before {
  content: '\\e93f';
}

.c2.icon-youtube:before {
  content: '\\e944';
}

.c2.icon-facebookOutline:before {
  content: '\\e91e';
}

.c2.icon-instagramOutline:before {
  content: '\\e924';
}

.c2.icon-spotifyOutline:before {
  content: '\\e927';
}

.c2.icon-twitterOutline:before {
  content: '\\e92d';
}

.c2.icon-youtubeOutline:before {
  content: '\\e92e';
}

.c2.icon-caretUp:before {
  content: '\\e932';
}

.c2.icon-caretDown:before {
  content: '\\e919';
}

.c2.icon-caretLeft:before {
  content: '\\e933';
}

.c2.icon-caretRight:before {
  content: '\\e934';
}

.c2.icon-caretCircleDown:before {
  content: '\\e91b';
}

.c2.icon-chevronLeft:before {
  content: '\\e92a';
}

.c2.icon-chevronRight:before {
  content: '\\e93c';
}

.c2.icon-chevronUp:before {
  content: '\\e93b';
}

.c2.icon-chevronDown:before {
  content: '\\e931';
}

.c2.icon-moveVertical:before {
  content: '\\e92c';
}

.c2.icon-check:before {
  content: '\\e938';
}

.c2.icon-checkCircle:before {
  content: '\\e902';
}

.c2.icon-alertCircle:before {
  content: '\\e91a';
}

.c2.icon-alertTriangle:before {
  content: '\\e901';
}

.c2.icon-calendar:before {
  content: '\\e917';
}

.c2.icon-time:before {
  content: '\\e93a';
}

.c2.icon-timeHand:before {
  content: '\\e939';
}

.c2.icon-calendarAdd:before {
  content: '\\e93d';
}

.c2.icon-byob:before {
  content: '\\e908';
}

.c2.icon-close:before {
  content: '\\e918';
}

.c2.icon-commercial:before {
  content: '\\e915';
}

.c2.icon-disabled:before {
  content: '\\e910';
}

.c2.icon-eye:before {
  content: '\\e91c';
}

.c2.icon-video:before {
  content: '\\e921';
}

.c2.icon-happyMask:before {
  content: '\\e922';
}

.c2.icon-helpCircleOutline:before {
  content: '\\e925';
}

.c2.icon-ellipsisHorizontal:before {
  content: '\\e92b';
}

.c2.icon-no:before {
  content: '\\e92f';
}

.c2.icon-pet:before {
  content: '\\e907';
}

.c2.icon-readGlasses:before {
  content: '\\e930';
}

.c2.icon-residential:before {
  content: '\\e916';
}

.c2.icon-tree:before {
  content: '\\e906';
}

.c2.icon-heartPulse:before {
  content: '\\e91f';
}

.c2.icon-homeWorld:before {
  content: '\\e923';
}

.c2.icon-travelWorld:before {
  content: '\\e928';
}

.c2.icon-musicTicket:before {
  content: '\\e926';
}

.c2.icon-musicLove:before {
  content: '\\e91d';
}

.c2.icon-plus:before {
  content: '\\e935';
}

.c2.icon-padlock:before {
  content: '\\e936';
}

.c2.icon-search:before {
  content: '\\e830';
}

.c2.icon-music:before {
  content: '\\e942';
}

.c2.icon-mail:before {
  content: '\\e941';
}

.c2.icon-expandHorizontal:before {
  content: '\\e93e';
}

.c2.icon-reset:before {
  content: '\\e912';
}

.c2.icon-pencil:before {
  content: '\\e904';
}

.c2.icon-trashCan:before {
  content: '\\e94a';
}

.c2.icon-arrowUp:before {
  content: '\\e913';
}

.c2.icon-arrowLeft:before {
  content: '\\e949';
}

.c2.icon-info:before {
  content: '\\e90d';
}

.c2.icon-infoOutline:before {
  content: '\\e948';
}

.c2.icon-feedback:before {
  content: '\\e905';
}

.c2.icon-feedbackOutline:before {
  content: '\\e947';
}

.c2.icon-media:before {
  content: '\\e909';
}

.c2.icon-mediaOutline:before {
  content: '\\e946';
}

.c2.icon-concert:before {
  content: '\\e90a';
}

.c2.icon-concertOutline:before {
  content: '\\e945';
}

.c2.icon-map:before {
  content: '\\e914';
}

.c2.icon-grid:before {
  content: '\\e900';
}

.c2.icon-photos:before {
  content: '\\e911';
}

.c2.icon-artistParking:before {
  content: '\\e90f';
}

.c2.icon-guestParking:before {
  content: '\\e90e';
}

.c2.icon-eventSetUp:before {
  content: '\\e90c';
}

.c2.icon-fanExperience:before {
  content: '\\e90b';
}

.c2.icon-rock:before {
  content: '\\e900';
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<Component
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "default": "2px",
        "input": "2px",
        "large": "8px",
      },
      "colors": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueChristmas": "#3498DB",
        "blueMagic": "#34495E",
        "blueSmoke": "#949494",
        "brownSugar": "#9B525F",
        "checkBerry": "#98035A",
        "darkSmoke": "#767676",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green800": "#008736",
        "green900": "#006721",
        "greenGarden": "#19BC9B",
        "greenOnions": "#8BC34B",
        "macyGrey": "#DBDBDB",
        "mellowYellow": "#F1C410",
        "orangeCrush": "#F39C12",
        "paintItBlack": "#333333",
        "peaches": "#F99973",
        "pinkTriangle": "#E91E63",
        "primary": "#039842",
        "purpleHaze": "#3F51B5",
        "purplePills": "#9B59B6",
        "redRedWine": "#EF4423",
        "royalBlood": "#FF5000",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
        "yellowSubmarine": "#FFD426",
      },
      "dimensions": Object {
        "button": Object {
          "default": Object {
            "height": "44px",
            "padding": "0 25px",
          },
          "small": Object {
            "height": "40px",
            "padding": "0 15px",
          },
        },
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
        "navbarHeight": Object {
          "lg": "84px",
          "md": "74px",
          "xs": "62px",
        },
        "radioBox": "20px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "formGroupLabel": "14px",
        "h1": Object {
          "lg": "48px",
          "md": "34px",
          "xs": "24px",
        },
        "h2": Object {
          "lg": "34px",
          "md": "24px",
          "xs": "20px",
        },
        "h3": Object {
          "lg": "24px",
          "md": "20px",
          "xs": "16px",
        },
        "h4": Object {
          "lg": "20px",
          "md": "16px",
          "xs": "16px",
        },
        "h5": Object {
          "lg": "16px",
          "md": "14px",
          "xs": "14px",
        },
        "h6": Object {
          "lg": "14px",
          "md": "14px",
          "xs": "14px",
        },
        "overline": "10px",
        "title": Object {
          "lg": "54px",
          "md": "38px",
          "xs": "26px",
        },
      },
      "fonts": Object {
        "regular": "Open Sans",
      },
      "media": Object {
        "lg": [Function],
        "md": [Function],
        "sm": [Function],
        "xl": [Function],
        "xs": [Function],
      },
      "ruler": Array [
        0,
        4,
        8,
        12,
        16,
        20,
        24,
        28,
        32,
        36,
        40,
        44,
        48,
        52,
        56,
        60,
        64,
        68,
        72,
        76,
        80,
      ],
      "utils": Object {
        "transition": [Function],
      },
      "zIndex": Object {
        "navbar": 100,
      },
    }
  }
>
  <Pagination
    currentPage={3}
    onPageChange={[MockFunction]}
    totalRecords={80}
  >
    <styled.div
      data-qaid=""
    >
      <div
        className="c0"
        data-qaid=""
      >
        <Component
          data-qaid="-prev"
          disabled={false}
          onClick={[Function]}
        >
          <styled.div
            as="button"
            clickable={true}
            data-qaid="-prev"
            disabled={false}
            onClick={[Function]}
          >
            <button
              className="c1"
              data-qaid="-prev"
              disabled={false}
              onClick={[Function]}
            >
              <Icon
                name="chevronLeft"
              >
                <styled.i
                  className="icon-chevronLeft "
                >
                  <i
                    className="c2 icon-chevronLeft "
                  />
                </styled.i>
              </Icon>
            </button>
          </styled.div>
        </Component>
        <Compact
          currentPage={3}
          onClickPage={[Function]}
          qaId=""
          totalPages={10}
        >
          <Pages
            currentPage={3}
            displayPages={5}
            onClick={[Function]}
            qaId=""
          >
            <PageBubble
              isActive={false}
              key="1"
              onClick={[Function]}
              page={1}
              qaId=""
            >
              <Styled(styled.div)
                as="button"
                clickable={true}
                data-qaid="-page-1"
                onClick={[Function]}
              >
                <button
                  className="c1 c3"
                  data-qaid="-page-1"
                  onClick={[Function]}
                >
                  1
                </button>
              </Styled(styled.div)>
            </PageBubble>
            <PageBubble
              isActive={false}
              key="2"
              onClick={[Function]}
              page={2}
              qaId=""
            >
              <Styled(styled.div)
                as="button"
                clickable={true}
                data-qaid="-page-2"
                onClick={[Function]}
              >
                <button
                  className="c1 c3"
                  data-qaid="-page-2"
                  onClick={[Function]}
                >
                  2
                </button>
              </Styled(styled.div)>
            </PageBubble>
            <PageBubble
              isActive={true}
              key="3"
              onClick={[Function]}
              page={3}
              qaId=""
            >
              <Styled(styled.div)
                color="primary"
              >
                <div
                  className="c4 c3"
                  color="primary"
                >
                  3
                </div>
              </Styled(styled.div)>
            </PageBubble>
            <PageBubble
              isActive={false}
              key="4"
              onClick={[Function]}
              page={4}
              qaId=""
            >
              <Styled(styled.div)
                as="button"
                clickable={true}
                data-qaid="-page-4"
                onClick={[Function]}
              >
                <button
                  className="c1 c3"
                  data-qaid="-page-4"
                  onClick={[Function]}
                >
                  4
                </button>
              </Styled(styled.div)>
            </PageBubble>
            <PageBubble
              isActive={false}
              key="5"
              onClick={[Function]}
              page={5}
              qaId=""
            >
              <Styled(styled.div)
                as="button"
                clickable={true}
                data-qaid="-page-5"
                onClick={[Function]}
              >
                <button
                  className="c1 c3"
                  data-qaid="-page-5"
                  onClick={[Function]}
                >
                  5
                </button>
              </Styled(styled.div)>
            </PageBubble>
          </Pages>
          <Ellipsis>
            <styled.div>
              <div
                className="c5"
              >
                <Icon
                  name="ellipsisHorizontal"
                >
                  <styled.i
                    className="icon-ellipsisHorizontal "
                  >
                    <i
                      className="c2 icon-ellipsisHorizontal "
                    />
                  </styled.i>
                </Icon>
              </div>
            </styled.div>
          </Ellipsis>
          <PageBubble
            isActive={false}
            onClick={[Function]}
            page={10}
            qaId=""
          >
            <Styled(styled.div)
              as="button"
              clickable={true}
              data-qaid="-page-10"
              onClick={[Function]}
            >
              <button
                className="c1 c3"
                data-qaid="-page-10"
                onClick={[Function]}
              >
                10
              </button>
            </Styled(styled.div)>
          </PageBubble>
        </Compact>
        <Component
          data-qaid="-next"
          disabled={false}
          onClick={[Function]}
        >
          <styled.div
            as="button"
            clickable={true}
            data-qaid="-next"
            disabled={false}
            onClick={[Function]}
          >
            <button
              className="c1"
              data-qaid="-next"
              disabled={false}
              onClick={[Function]}
            >
              <Icon
                name="chevronRight"
              >
                <styled.i
                  className="icon-chevronRight "
                >
                  <i
                    className="c2 icon-chevronRight "
                  />
                </styled.i>
              </Icon>
            </button>
          </styled.div>
        </Component>
      </div>
    </styled.div>
  </Pagination>
</Component>
`;
